src/test/scala/io/iohk/ethereum/domain/BlockchainReaderSpec.scala

```diff
@@ -0,0 +1,19 @@
+package io.iohk.ethereum.domain
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import io.iohk.ethereum.Fixtures
+import io.iohk.ethereum.blockchain.sync.EphemBlockchainTestSetup
+
+class BlockchainReaderSpec extends AnyFlatSpec with Matchers {
+
+  "BlockchainReader" should "be able to get the best block after it was stored by BlockchainWriter" in new EphemBlockchainTestSetup {
+    val validBlock = Fixtures.Blocks.ValidBlock.block
```

Comment:
proposal: it looks like this could be transformed into a more generic property based test (replace the fixture.block with a generated one), unless it's hard to generate a valid block usable by the blockwriter.

Reply:
It could yes. I'm not really sure what would be the added value of a property test there though as is it quite simple (just storing and retrieving and no real logic involved). Maybe we should have a rule of thumb on what would need a property test.

Reply:
Could any of the randomized data (or size of the data) impact the property you want to test? Maybe edge cases?

Reply:
I added a property test instead